src/core/mempool.rs

```diff
@@ -241,10 +242,16 @@ pub struct MemPoolTxInfo {
     pub metadata: MemPoolTxMetadata,
 }
 
-#[derive(Debug, PartialEq, Clone)]
-pub enum MemPoolTxInfoPartial {
-    NeedsNonces { addrs_needed: Vec<StacksAddress> },
-    HasNonces(MemPoolTxInfo),
+/// This class is a minimal version of `MemPoolTxInfo`. It contains
+/// just enough information to 1) filter by nonce readiness, 2) sort by fee rate.
+#[derive(Debug, Clone)]
+pub struct MemPoolTxInfoPartial {
```

Comment:
Are `origin_nonce` and `sponsor_nonce` still used in this struct?  Are they still used anywhere in the mempool?

If not, then can you add a comment to indicate that they'll be removed in the near future, and open an issue?  No need to block the PR on it, given that we're pressed for time.